app/components/ActionButtons.tsx

```diff
@@ -1,25 +1,54 @@
 import React from 'react';
 import { GameState } from '../types/game';
+import { hasFlor } from '../utils/gameUtils';
 
 interface ActionButtonsProps {
   phase: GameState['phase'];
   trucoState: GameState['trucoState'];
+  envidoState: GameState['envidoState'];
+  playedCards: GameState['playedCards'];
+  humanPlayedCard: GameState['humanPlayedCard'];
   nextTurnProgress: number;
+  humanCards: GameState['humanCards'];
+  aiCards: GameState['aiCards'];
+  muestraCard: GameState['muestraCard'];
   onNextRound: () => void;
   onNextTurn: () => void;
   onTruco: () => void;
+  onEnvido: () => void;
   onTrucoResponse: (action: 'accept' | 'reject' | 'escalate') => void;
+  onEnvidoResponse: (action: 'accept' | 'reject') => void;
 }
 
 const ActionButtons: React.FC<ActionButtonsProps> = ({
   phase,
   trucoState,
+  envidoState,
+  playedCards,
+  humanPlayedCard,
   nextTurnProgress,
+  humanCards,
+  aiCards,
+  muestraCard,
   onNextRound,
   onNextTurn,
   onTruco,
-  onTrucoResponse
-}) => (
+  onEnvido,
+  onTrucoResponse,
+  onEnvidoResponse
+}) => {
+  // Helper function to determine if Envido button should be shown
+  const canShowEnvidoButton = () => {
+    const isHumanTurn = phase.type === 'HUMAN_TURN';
+    const isEarlyInRound = playedCards.length === 0 || (playedCards.length === 1 && humanPlayedCard === null);
+    const noEnvidoCalled = envidoState.type === 'NONE';
+    const noTrucoCalled = trucoState.type === 'NONE';
+    const noFlor = !hasFlor(humanCards, muestraCard) && !hasFlor(aiCards, muestraCard);
```

Comment:
I've found there is an issue with `hasFlor(aiCards, muestraCard)` when the AI already played a card, because the played card is not longer part of `aiCards`. So, the flor calculation could be wrong. `hasFlor` should be invoked always with the original 3 cards of the player.